packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -93,6 +167,39 @@ export class RichTextEditor extends FoundationElement {
     public override disconnectedCallback(): void {
         super.disconnectedCallback();
         this.unbindEditorTransactionEvent();
+        this.unbindEditorUpdateEvent();
+        this.resizeObserver?.disconnect();
+    }
+
+    /**
+     * @internal
+     */
+    public disabledChanged(): void {
+        if (this.tiptapEditor) {
+            this.tiptapEditor.setEditable(!this.disabled);
+            this.editor.setAttribute(
+                'aria-disabled',
+                this.disabled ? 'true' : 'false'
+            );
```

Comment:
While it makes sense absolutely to set `aria-disabled` to `true` when `disabled` attribute changes, I am wondering whether custom slot contents will still be left enabled?

Do we also need to disable slot content in template then (or) remove setting this attribute? It will be good to get thoughts from Nimble team. Since this is a minor one, we can keep it open and get directions from Nimble team's expertise.

Reply:
Yes, the custom slot elements will not be disabled when we set the `disabled` attribute. It is the responsibility of the clients to disable the slot contents. It is the same in the [nimble-text-field](https://nimble.ni.dev/storybook/?path=/docs/components-text-area--docs) where the slot element will not be disabled based on the `disabled` attribute. Also, we clearly mentioned in the storybook docs the same way as the text field mentioned for slot elements.
![image](https://github.com/ni/nimble/assets/123377523/3ea7f2e5-8dfe-4607-8503-696011ec9bab)

As per the [MDN reference for aria-disabled](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled), the custom `<div>` control used in the editor should set the `aria-disabled` value.
![image](https://github.com/ni/nimble/assets/123377523/6bd5dd10-deab-45ee-a8ee-c314e2399bc8)